### DISCUSSION TITLE
useForm with promise?

### DISCUSSION THREAD

**Discussion by torian257x** [react-hook-form/react-hook-form#2303] [Q&A] [answered]
for:
https://react-hook-form.com/advanced-usage#FieldArrays

what if I want to load the given values via axios.get or the like

afaik

```
  const { register, control, handleSubmit, reset } = useForm({
    defaultValues: {
      test: [{ firstName: "Bill2222", lastName: "Luo" }]
    }
  });
```

is the way to set the values that are already set

but... I need to retrieve those values from an API.

How?

**Comment by bluebill1049** [comment, marked as answer]
useEffect with reset :)

**Reply by torian257x** [reply]
https://react-hook-form.com/api/#reset
